test/es6/ES6Promise.js

```diff
@@ -91,10 +91,10 @@ var tests = [
     {
         name: "Promise constructor throwing behavior",
         body: function () {
-            assert.throws(function() { Promise.call(); }, TypeError, "Promise throws when not called as a new expression with no this parameter", "Promise: 'this' is not a Promise object");
-            assert.throws(function() { Promise.call(undefined); }, TypeError, "Promise throws when not called as a new expression if the this parameter is undefined", "Promise: 'this' is not a Promise object");
-            assert.throws(function() { Promise.call(null); }, TypeError, "Promise throws when not called as a new expression if the this parameter is null", "Promise: 'this' is not a Promise object");
-            assert.throws(function() { Promise.call({}); }, TypeError, "Promise throws when not called as a new expression if the this parameter is not a promise", "Promise: 'this' is not a Promise object");
+            assert.throws(function() { Promise.call(); }, TypeError, "Promise throws when not called as a new expression with no this parameter", "'new.target' is undefined");

```

Comment:
The meaning of the message is "Promise cannot be called without the 'new' keyword" but spec uses the case where new.target is undefined to indicate this. I don't think there is a way for new.target to actually be undefined if we do have the new flag, though. Even Reflect.construct(Promise,[],undefined) throws. I'll change the error message here.